src/libstore/sqlite.hh

```diff
@@ -96,10 +96,20 @@ struct SQLiteTxn
 };
 
 
-MakeError(SQLiteError, Error);
-MakeError(SQLiteBusy, SQLiteError);
+struct SQLiteError : Error
+{
+    const char *path;
+    int errNo, extendedErrNo;
+
+    template<typename... Args>
+    [[noreturn]] static void throw_(sqlite3 * db, const std::string & fs, const Args & ... args);
```

Comment:
Unless I’m missing something, this is very fragile since the definition of the templated function isn’t in the header − so only the instances of the template that happen to be used in `sqlite.cc` will be available outside (which happens to be the case because the only usages outside of `sqlite.cc` seem to be whith `args` being a single string)

Reply:
Oh good catch!

Reply:
Fixed this.

Reply:
Fixed it an even better way!